File: quiz-nodejs-engine/src/models/QuizResponse.ts
import 'reflect-metadata';
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToOne,
  JoinColumn,
  CreateDateColumn,
} from 'typeorm';
import { QuizSession } from './QuizSession';
import { Question } from './Question';

@Entity('quiz_responses')
export class QuizResponse {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ type: 'varchar', length: 36 })
  sessionId!: string;

  @Column({ type: 'integer' })
  questionId!: number;

  @Column({ type: 'varchar', length: 1 })
  userAnswer!: string;

  @Column({ type: 'integer', default: 0 })
  isCorrect!: number;

  @Column({ type: 'integer', nullable: true })
  timeTakenSeconds?: number;

  @CreateDateColumn()
  answeredAt!: Date;

  @ManyToOne(() => QuizSession, (session) => session.responses, { lazy: true })
  @JoinColumn({ name: 'sessionId' })
  session!: Promise<QuizSession>;

  @ManyToOne(() => Question, (question) => question.responses, { lazy: true })
  @JoinColumn({ name: 'questionId' })
  question!: Promise<Question>;
}

File: quiz-engine-springboot/src/main/java/com/quizengine/entity/QuizResponse.java
package com.quizengine.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "quiz_responses")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuizResponse {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "session_id", referencedColumnName = "sessionId")
    private QuizSession session;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id", referencedColumnName = "id")
    private Question question;

    @Column(nullable = false)
    private String userAnswer;

    @Column
    @Builder.Default
    private Integer isCorrect = 0;

    @Column
    private Integer timeTakenSeconds;
}
